File: src/main/java/io/github/chase22/infrastructure/server/ApiServer.java
package io.github.chase22.infrastructure.server;

import io.github.chase22.infrastructure.server.handler.HandlerChain;
import io.undertow.Undertow;

public class ApiServer {

    private final Undertow server;

    ApiServer(int port, HandlerChain handlerChain) {
        server = Undertow.builder().addHttpListener(port, "localhost")
                .setHandler(handlerChain.getHandler())
                .build();
    }

    public void start() throws Exception {
        server.start();
    }

    public void stop() throws Exception {
        server.stop();
    }
}

File: src/main/kotlin/io/github/chase22/infrastructure/server/ApiServer.kt
package io.github.chase22.infrastructure.server

import io.github.chase22.infrastructure.server.handler.HandlerChain
import io.undertow.Undertow

class ApiServer internal constructor(port: Int, handlerChain: HandlerChain) {

    private val server: Undertow = Undertow
            .builder()
            .addHttpListener(port, "localhost")
            .setHandler(handlerChain.handler)
            .build()

    fun start() {
        server.start()
    }

    fun stop() {
        server.stop()
    }
}
